app/generic/verifier/verifier.py

```diff
@@ -27,13 +31,129 @@ class VerifierFacade(Enum):
     v20 = VerifierV2
 
 
-def __get_verifier_by_version(protocol_version: str) -> Verifier:
-    if protocol_version == "v1":
+def __get_verifier_by_version(version_candidate: str) -> Verifier:
+    if version_candidate == "v1" or version_candidate.startswith("v1-"):
         return VerifierFacade.v10.value
-    elif protocol_version == "v2":
+    elif version_candidate == "v2" or version_candidate.startswith("v2-"):
         return VerifierFacade.v20.value
     else:
-        raise ValueError(f"Unknown protocol version {protocol_version}")
+        raise ValueError(f"Unknown protocol version {version_candidate}")
+
+
+@router.post("/credentials")
+async def get_credentials_for_request(
+    proof_request: ProofRequestGeneric,
+    aries_controller: AcaPyClient = Depends(agent_selector),
+) -> List[IndyCredPrecis]:
+    """
+     Get matching credentials for  presentation exchange
+
+     Parameters:
+     ----------
+    proof_request: ProofRequestGeneric
+         The proof request object
+
+     Returns:
+     --------
+     presentation_exchange_list: [IndyCredPrecis]
+         The list of Indy presentation credentials
+    """
+    try:
+        prover = __get_verifier_by_version(
+            version_candidate=proof_request.protocol_version
```

Comment:
We can extract the `protocol_version` from the `proof_id`. As now it is possible to pass `proof_id=v1-xxx` and `protocol_version='v2'`, which can lead to weird inconsistencies